src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -70,13 +79,35 @@ private String getParameter(HttpServletRequest request, String name, String defa
   }
 
   /**
-  * Converts a UserAuth object into a JSON string using the Gson library.
-  */
+   * Converts a UserAuth object into a JSON string using the Gson library.
+   */
   private String convertToJson(UserAuth userAuth) {
     Gson gson = new Gson();
     String json = gson.toJson(userAuth);
     return json;
   }
+  
+  /**
+   * If the user is not in database already, add them.
+   */
+  private void addUserToDatabase(String email) {
+    // Query database to see if User has already been added.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query(User.USER);
+    PreparedQuery results = datastore.prepare(query);
+
+    // If user is already in database, return.
+    for (Entity entity : results.asIterable()) {
+      if (entity.getProperty(User.USER_EMAIL).equals(email)) {
+        return;
+      }
+    }
+
+    // User is not in database, create and add user object.
+    User user = new User(email);
```

Comment:
That's a good point; I'll add a method just to create the Entity instead of creating an Entity from a User object. (I'll still leave the original method though, as I think it is useful.)

Reply:
Actually, I checked and on my next CL, I need to build the User as well, so I'm going to keep the current structure.